scheduler/pkg/agent/repository/mlserver/settings.go

```diff
@@ -0,0 +1,182 @@
+package mlserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/seldonio/seldon-core/apis/go/v2/mlops/scheduler"
+)
+
+var (
+	sklearnExtensions  = []string{"joblib", "pkl", "pickle"}
+	xgboostExtensions  = []string{"json", "bst"}
+	lightgbmExtensions = []string{"bst"}
+)
+
+// Look at requirements and take first matching one to our known list of ones we can handle
+func createSettingsFile(path string, modelSpec *scheduler.ModelSpec) error {
+	modelSettings, err := getModelSettings(path, modelSpec)
+	if err != nil {
+		return err
+	}
+	return saveModelSettings(path, modelSettings)
+}
+
+func getModelSettings(path string, modelSpec *scheduler.ModelSpec) (*ModelSettings, error) {
+	for _, requirement := range modelSpec.Requirements {
```

Comment:
`Requirements` is a free-form text for labels. I am not sure if we should rely on them exactly. Perhaps consider having a type for the model? maybe the `cli` can help populating the type?

Reply:
This is a bigger discussion as all scheduling is based on matching whats specified in the requirements in the model with the capabilities of the server when started and defined in the ServerConfig

Reply:
I agree but the servers / models can have any set of labels for this. So not necessarily that these tags remain the same in the future and for customer. 

Yeah we could argue that it is out of scope of this PR